### DISCUSSION TITLE
Improving DX when working with `run-script` and workspaces

### DISCUSSION THREAD

**Discussion by cawa-93** [npm/rfcs#350] [RRFC]
When you run a command in workspaces, it is not obvious which log comes from which workspace.

For example, I have a monorepo with a certain number of packages. Each package has a "build" script. When I run `npm run build --ws` I get the following log:

```
> build
> node build.js

... some logs


> build
> node build.js

ERROR: ...


> build
> node build.js

... some logs
```
From the standard output it is absolutely not clear which log belongs to which workspace.

It would be much more convenient if npm grouped any log by workspaces and mark of them:
```
workspace-a:

    > build
    > node build.js

    ... some logs


workspace-b:

    > build
    > node build.js

    ERROR: ...


workspace-c:

    > build
    > node build.js

    ... some logs
```

**Comment by ljharb** [comment]
Alternatively, each line of the output could be prefixed with `[workspace-N]`, kind of like how lerna does it?

**Reply by cawa-93** [reply]
I have never used lerna. But it seems to me that in this case, logging in will be very cumbersome? In addition, the simple grouping is easier to implement. Somehow like [this](https://github.com/npm/cli/compare/latest...cawa-93:patch-1).

**Reply by ljharb** [reply]
Many log parsers operate line-by-line, and the "grouping" method would break those.

**Reply by cawa-93** [reply]
Hmm ... I didn't think about that
